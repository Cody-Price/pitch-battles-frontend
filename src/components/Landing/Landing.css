.game-title {
  margin-top: 2.5rem;
  font-family: "Scriptorium";
  font-size: 3rem;
  color: #e3e8d9;
  text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
}

.logo {
  width: 80px;
  height: 120px;
  margin: 2rem 1rem;
  z-index: 101;
  background-size: 324px 120px;
  animation: bard 1s steps(4) infinite;
  image-rendering: pixelated;
  position: relative;
}

.login-signup-container {
  width: 341px;
  height: 431px;
  margin: 0 auto;
  background: url(./landingframe3.png);
  background-size: cover;
  image-rendering: pixelated;
  position: relative;
  overflow: hidden;
  z-index: 100;
}

.landing-page {
  position: relative;
  overflow: hidden;
  height: 100vh;
}

.landing-page-header {
  margin-top: 1rem;
  position: relative;
  z-index: 10;
  display: flex;
  justify-content: center;
}

.landing-version-number {
  position: absolute;
  right: 2px;
  bottom: 0px;
  font-family: "chronotype";
  color: #f2f2f2;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.9);
  margin: 0;
}

.toggle-dev-message-button {
  background: url("./images/dev-message-btn.png");

  position: absolute;
  right: 0;
  top: 0;
  margin: 1rem;
  height: 27px;
  width: 15px;
  image-rendering: pixelated;
  background-size: cover;
  border: none;
  z-index: 1000;
  padding: 0;
}

.toggle-dev-message-button:focus {
  outline: none;
}

.toggle-dev-message-button:hover {
  cursor: pointer;
}

@keyframes bard {
  100% {
    background-position: -324px;
  }
}

@media (max-width: 700px) {
  .logo {
    margin-top: 1rem;
    margin-bottom: 1rem;
  }
  .game-title {
    margin-top: 1.5rem;
    margin-bottom: 1rem;
  }
}
