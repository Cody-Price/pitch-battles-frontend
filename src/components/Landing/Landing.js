import React, { Component } from 'react';
import Login from '../Login/Login';
import Signup from '../Signup/Singup';

class Landing extends Component {
	constructor() {
		super();
		this.state = {
			login: true,
			signup: false
		}
	}

	landingChange = () => {
		this.setState({login: !this.state.login, signup: !this.state.signup})
	}

	render() {
		return (
			<main>
				<header>
					<h1>Pitch Battles</h1>
					<div className='logo' />
				</header>
				<section className='login-signup-container'>
					<Login landingChange={this.landingChange} status={this.state.login} />
					<Signup landingChange={this.landingChange} status={this.state.signup} />
				</section>
			</main>
		)
	}
}

export default Landing;
