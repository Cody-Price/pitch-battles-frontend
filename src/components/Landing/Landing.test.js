import React from "react";
import ReactDOM from "react-dom";
import Landing from "./Landing";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Landing", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Landing />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("landingChange", () => {
    it("should call setState", () => {
      const expected = {
        forgotPassword: false,
        login: false,
        resetPassword: false,
        signup: true,
        showDevMessage: false
      };
      wrapper.instance().landingChange();

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("forgotPasswordScreen", () => {
    it("should call setState", () => {
      const expected = {
        forgotPassword: true,
        login: false,
        resetPassword: false,
        signup: false,
        showDevMessage: false
      };
      wrapper.instance().forgotPasswordScreen();

      expect(wrapper.state()).toEqual(expected);
    });
  });
});
