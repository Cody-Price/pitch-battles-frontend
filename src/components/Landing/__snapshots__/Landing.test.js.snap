// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
<main
  className="landing-page"
>
  <header
    className="landing-page-header"
  >
    <div
      className="logo"
    />
    <h1
      className="game-title"
    >
      Pitch
      <br />
       Battles
    </h1>
  </header>
  <section
    className="login-signup-container"
  >
    <Login
      forgotPasswordScreen={[Function]}
      landingChange={[Function]}
      status={true}
    />
    <Signup
      landingChange={[Function]}
      status={false}
    />
    <ForgotPassword
      backToLogin={[Function]}
      status={false}
    />
  </section>
</main>
`;
